pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -169,6 +166,29 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, updateErr
 	}
 
+	if err := r.upsertManagedObjects(ctx, reconcilerRef, currentRS, rs); err != nil {
+		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
+		return controllerruntime.Result{}, err
+	}
+
+	metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(nil), start)
+	return controllerruntime.Result{}, nil
+}
+
+func (r *RepoSyncReconciler) upsertManagedObjects(ctx context.Context, reconcilerRef types.NamespacedName, currentRS, rs *v1beta1.RepoSync) error {
+	rsRef := client.ObjectKeyFromObject(rs)
+	r.logger(ctx).V(3).Info("Reconciling managed objects")
+
+	// If Reconciling condition does not exist, set it to "Initialization".
+	// Avoid updating the timestamps on every retry.
+	cond := reposync.GetCondition(rs.Status.Conditions, v1beta1.RepoSyncReconciling)
+	if cond == nil {
+		reposync.SetReconciling(rs, "Initialization", "Setup in progress")
```

Comment:
Looks like this is new - I'm curious what is the goal?

Looks like this `Initialization` status will exist for a new RSync until the reconciler Deployment is created?

Reply:
Ah yeah. I added this as the opposite of Finalization. It means the controller has validated the spec but the deployment hasn’t been created yet. 

I’ll rip it out and put it in a follow up with unit tests for it.

Reply:
Thank you, that will help with isolating/testing the change

Reply:
Pulled out. Will make a follow-up with it once this is merged.